### PR TITLE
m365_defender: add geoip processing to {source,destination}.ip

### LINKED ISSUE #8891: [M365 Defender] Expand use of GeoIP Processor
The Microsoft M365 Defender integration does not currently apply the geoip processor to the [event.devices pipeline](https://github.com/elastic/integrations/blob/main/packages/m365_defender/data_stream/event/elasticsearch/ingest_pipeline/pipeline_device.yml) to the `source.ip` or `dest.ip` fields. The processor is limited to `host.ip`. This results in users having to add logic to the @custom pipeline in order to geolocalize these import fields. 

Can we please add the geoip processor to both the `source.ip` and `dest.ip` fields within this pipeline. 

We should remove the manual renaming of the source geo (city name and iso code) then, so that the geo processor can assign something that matches.

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

See title.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

- Closes #8891

### CODE REVIEW

File: packages/m365_defender/changelog.yml [hunk 1]
```diff
@@ -1,7 +1,12 @@
 # newer versions go on top
+- version: "2.6.0"
+  changes:
+    - description: Add geoip processing to source and destination IP fields.
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/1
```
Comment (kcreddy) [inline packages/m365_defender/changelog.yml:6]: pr number

**Review by bhapas — APPROVED** [review]
LGTM. Please merge once the PR number in changelog is fixed.

File: packages/m365_defender/changelog.yml (hunk 1 above)
Comment (efd6) [reply packages/m365_defender/changelog.yml:6]: Thanks

**Review by kcreddy — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**